app.js

```diff
@@ -60,15 +60,51 @@ app.post('/api/addUserRequest', function (req, res) {
 
   // process entry
   if (checkParams.valid === true) {
-    var pendingRequest = req.body;
-    pendingRequest["time_submitted"] = new Date();
-    pendingRequest["helped"] = false;
 
-    db.child("requests").push(req.body, function () {
-      res.status(200).end();
+    var hasDuplicateCase = false;
+
+    // check if a request with the same contact info has not already been submitted
+    db.child("cases").once("value", function (snapshot) {
+      if (snapshot.val() !== null) {
+        for (var k in snapshot.val()) {
+          var numberPattern = /\d+/g;
+
+          if (req.body.contactMethod == "SMS") {
+
+            if ((snapshot.val()[k].contact).match(numberPattern).join("").substr(-10) == (req.body.contact).match(numberPattern).join("").substr(-10)) {
+              hasDuplicateCase = true;
+              break;
+            }
+          } else if (req.body.contactMethod == "Email" && snapshot.val()[k].contact == req.body.contact) {
+            hasDuplicateCase = true;
+            break;
+          }
+        }
+
+        if (hasDuplicateCase == false) {
```

Comment:
Replace this with:
```javascript
if (hasDuplicateCase === true) {
    return;
}
```
That should work since we are in the top-level response function.